### ISSUE TITLE
validated-non-primitive-property-needs-type-decorator doesn't work on Arrays

### ISSUE THREAD

**Issue by iddan-flycode** [darraghoriordan/eslint-plugin-nestjs-typed#15]
```typescript
// valid
class Example {
    @ApiProperty({ isArray: true })
    @ValidateNested({ each: true })
    // @Type(() => Foo)
    repositories!: Foo[];
}

// invalid
class Example {
    @ApiProperty()
    @ValidateNested()
    // @Type(() => Foo)
    repositories!: Foo;
}
```

**Comment by darraghoriordan** [comment]
I had a look at this and I wrote it so it expect @IsArray() on the array property to check if the type should be present

i changed it to also support have validate nested only

fix in version 3.5.7

thanks for reporting

**Comment by iddan** [comment]
Thank you 🙏
